### PR TITLE
[WIP] unify config log options

### PR DESCRIPTION
**Pull request by butonic**
The current config option naming is a grown mess. This pr changes the logging options and introducas fallback code. I did not yet add logging, because during config reading the logger is not yet initialized ... might add a shutdown function to do that.

Any other logging related options I should add?

### CODE REVIEW

**Comment by butonic** [comment]
- [ ] add Documentation PR

File: lib/private/Config.php [hunk 1]
```diff
@@ -222,6 +227,54 @@ private function readData() {
 		}
 	}
 
+	private function updateConfigKeys(array $config) {
+		if (isset($config['log_type'])) {
+			$config['log.type'] = $config['log_type'];
```
Comment (phil-davis) [inline lib/private/Config.php]: Should the old keys be unset?

Comment (butonic) [inline lib/private/Config.php]: I was thinking of leaving them in, but yes, they should go ... still thinking about a way to best log usage of deprecated keys

Comment (phil-davis) [inline lib/private/Config.php]: Or have an upgrade step that updates old values in the config and write back to it, then forget that the old values ever existed.
There will also be the issue that in the comment text in the config there could be mentions of the old config keys (e.g. when copied from the sample config, or typed in by a sysadmin themselves explaining why they set something), and that will be misleading if the real key is updated but nit the comment.

**Review by PVince81 — CHANGES_REQUESTED** [review]
Please log a warning when keys cannot be updated because of readonly config.php

**Comment by PVince81** [comment]
continue or abandon ?

**Comment by PVince81** [comment]
I'd vote for abandon as this would need more work to provide key migrations, and there isn't much benefit so far in making this change

**Comment by butonic** [comment]
will change anyway ...

**Pull request closed without merging**